airflow/utils/file.py

```diff
@@ -17,32 +17,14 @@
 # specific language governing permissions and limitations
 # under the License.
 
-import errno
 import os
 import re
-import shutil
 import zipfile
-from contextlib import contextmanager
-from tempfile import mkdtemp
 from typing import Dict, List, Optional, Pattern
 
 from airflow import LoggingMixin, conf
 
 
```

Comment:
Can you maintain backward compatibility to make migrations easier for users? I think I don't need to take drastic steps and immediately delete this function without warning.  I think that changes should always be done in several stages:
-initial version
- initial version with a deprecation warning
- remove features.

We can make changes to Airflow 2.0 that are not backward compatible, but these should be individual exceptions, but not the only way to make changes.

Reply:
We can shorten this path if we suspect that users do not use this function, but I think that this does not apply in this case..  This PR introduces changes to 13 files, so this is a commonly used function.

Reply:
I added deprecation warning.